packages/aft/lib/src/commands/integration_test_command.dart

```diff
@@ -0,0 +1,232 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:aft/aft.dart';
+import 'package:aft/src/flutter_platform.dart';
+import 'package:aft/src/test_reports/test_report_folio.dart';
+import 'package:aft/src/test_reports/test_report_pass_fail.dart';
+import 'package:aft/src/test_reports/test_report_scored.dart';
+import 'package:aft/src/test_reports/test_score.dart';
+import 'package:aft/src/utils/constants.dart';
+import 'package:aft/src/utils/emphasize_text.dart';
+import 'package:aft/src/utils/get_active_devices.dart';
+import 'package:aft/src/utils/get_process_args.dart';
+import 'package:aft/src/utils/select_packages.dart';
+import 'package:interact/interact.dart';
+import 'package:path/path.dart' as p;
+
+import '../active_device.dart';
+import '../utils/execute_process.dart';
+
+typedef ResultIterator = void Function(
+  String result,
+  PackageInfo package,
+  String fileName,
+);
+const integrationTestPath = 'integration_test';
+
+/// Command to run integration tests.
+class IntegrationTestCommand extends AmplifyCommand {
+  IntegrationTestCommand();
+
+  @override
+  String get description =>
+      'Runs the appropriate integration test command for the given platform';
+
+  @override
+  String get name => 'integ';
+
+  final folio = TestReportFolio();
+
+  @override
+  Future<TestReportFolio> run() async {
+    final selectedDevice = await _selectDevice();
+    if (selectedDevice == null) {
+      stderr
+        ..writeln(
+          formatException('Valid device not selected.'),
+        )
+        ..writeln(
+          'Try running "flutter doctor" or "flutter devices" to ensure you have active devices.',
+        );
+      exit(1);
+    }
+
+    final packages = await allPackages;
+    final testablePackages = packages
+        .where(
+          (package) =>
+              package.platforms != null &&
+              package.platforms!.contains(selectedDevice.platform) &&
+              package.integTestDirectory != null,
+        )
+        .toList();
+
+    final selectedPackages = await selectPackages(testablePackages);
+
+    if (selectedPackages.isEmpty) {
+      stderr.writeln(formatException('You did not select any tests.'));
+      exit(1);
+    }
+
+    switch (selectedDevice.platform) {
+      case FlutterPlatform.android:
+      case FlutterPlatform.ios:
+      case FlutterPlatform.linux:
+      case FlutterPlatform.macos:
+      case FlutterPlatform.windows:
+        await _handleVM(
+          selectedPackages,
+          selectedDevice,
+        );
+        break;
+      case FlutterPlatform.web:
+        await _handleWeb(
+          selectedPackages,
+        );
+        break;
+      case null:
+        stderr.writeln(
```

Comment:
This pattern of `stderr.writeln(formatException(...)); exit(1);` should be extracted in a method. I usually call it `exitError`